experiments/views.py

```diff
@@ -41,3 +46,50 @@ def change_alternative(request, experiment_name, alternative_name):
 
     participant(request).set_alternative(experiment_name, alternative_name)
     return HttpResponse('OK')
+
+
+def decorate(decorators, patterns_rslt):
+    '''
+    Used to require 1..n decorators in any view returned by a url tree
+    Usage:
+      urlpatterns = required(func,patterns(...))
+      urlpatterns = required((func,func,func),patterns(...))
+    Note:
+      Use functools.partial to pass keyword params to the required
+      decorators. If you need to pass args you will have to write a
+      wrapper function.
+    Example:
+      from functools import partial
+      urlpatterns = required(
+          partial(login_required,login_url='/accounts/login/'),
+          patterns(...)
+      )
+    '''
+    if not hasattr(decorators,'__iter__'):
+        decorators = (decorators,)
+
+    return [
+        _wrap_instance__resolve(decorators, instance)
+        for instance in patterns_rslt
+    ]
+
+def _wrap_instance__resolve(wrapping_functions, instance):
+    if not hasattr(instance, 'resolve'):
+        return instance
+    
+    resolve = getattr(instance, 'resolve')
+    def _wrap_func_in_returned_resolver_match(*args,**kwargs):
+        rslt = resolve(*args,**kwargs)
+        if not hasattr(rslt, 'func'):
+            return rslt
+        
+        f = getattr(rslt, 'func')
+        for _f in reversed(wrapping_functions):
+            # @decorate the function from inner to outter
+            f = _f(f)
+
+        setattr(rslt, 'func', f)
+        return rslt
+
+    setattr(instance,'resolve',_wrap_func_in_returned_resolver_match)
+    return instance
```

Comment:
Check if it can be dropped.

Reply:
A leftover from a different approach. Devs can pivot too :D
Removed ✅